oracle-db/builtins.py

```diff
@@ -77,16 +81,19 @@ def __init__(self, db_config, *args, **kwargs):
         extra_keys = [key for key in db_config.keys() if key not in allowed_keys]
         for extra_key in extra_keys:
             db_config.pop(extra_key, None)
+
+        db_config.update({'engine':'oracle+cx_oracle'})
         db_config = self.db_config_schema(db_config)
 
         self.dsn = self.sdn_string.format(
-            engine=self.driver,
+            engine=db_config['engine'],
```

Comment:
both engine as well as database could be defined as constants in this file and not coming from configuration

Reply:
If database refers to `db=db_config['database']` so it is here the name of the database to connect to, so a user defined param in nature since it's defined by the user during oracle deployment